corehq/apps/zapier/views.py

```diff
@@ -86,3 +90,71 @@ def post(self, request, *args, **kwargs):
             return HttpResponse(status=400)
         delete_subscription_with_url(url)
         return HttpResponse('OK')
+
+
+class ZapierCreateCase(View):
+
+    urlname = 'zapier_create_case'
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, *args, **kwargs):
+        return super(ZapierCreateCase, self).dispatch(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        domain = request.GET.get('domain')
+        case_type = request.GET.get('case_type')
+        owner_id = request.GET.get('user_id')
+        properties = json.loads(request.body)
+        case_name = properties['case_name']
+        user_name = request.GET.get('user')
+
+        user = CommCareUser.get_by_username(user_name, domain)
+        couch_user = CouchUser.from_django_user(user)
+        if not couch_user.is_member_of(domain):
+            return HttpResponseForbidden("This user does not have access to this domain.")
+
+        del properties['case_name']
+
+        factory = CaseFactory(domain=domain)
+        factory.create_case(
+            case_type=case_type,
+            owner_id=owner_id,
+            case_name=case_name,
+            update=properties
+        )
+
+        return HttpResponse('OK')
+
+
+class ZapierUpdateCase(View):
+
+    urlname = 'zapier_update_case'
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, *args, **kwargs):
+        return super(ZapierUpdateCase, self).dispatch(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        domain = request.GET.get('domain')
+        case_type = request.GET.get('case_type')
+        user_name = request.GET.get('user')
+        properties = json.loads(request.body)
+        case_id = properties['case_id']
+
+        del properties['case_id']
+
+        user = CommCareUser.get_by_username(user_name, domain)
+        couch_user = CouchUser.from_django_user(user)
+        if not couch_user.is_member_of(domain):
+            return HttpResponseForbidden("This user does not have access to this domain.")
+
+        if case_id not in CaseAccessors(domain).get_case_ids_in_domain(case_type):
```

Comment:
I think this would be a big performance drain if there are a lot of cases in the domain.

It might be better to do something like:
`domain != CaseAccessors(domain).get_case(case_id).domain`
This will also throw a `CaseNotFound` exception if that case doesn't exist, which you should catch and send down as an HTTP error.

I think you should also validate if the case that you get from the DB is of the same case type as what the user said it was and send an error to Zapier if not.